### DISCUSSION TITLE
Support for old Xbox One S Controller Firmwares

### DISCUSSION THREAD

**Discussion by hipsofjw** [darthcloud/BlueRetro#304] [Ideas]
Hello!

I have two Xbox One S controllers that I'm keeping on the old firmware. The main reason is that updating them to the most recent BTLE firmwares makes them incompatible with my Brook adapters. This is not an issue for PS1, PS2, Dreamcast, and Saturn since I can make BlueRetro adapters for those consoles. However, it is an issue with the original Xbox and Xbox 360 since BlueRetro won't support these systems.

Would it be possible to include support in BlueRetro for the older pre-BTLE Xbox One S controller firmwares?

Anyway, thanks so much for this project! So far, I've made adapters for N64, PS2, PC-FX and 3DO. They work great!

**Comment by darthcloud** [comment]
I fully understand the pain but it's not really possible.

pre-BLE firmware got though various change that changed the buttons layout and I have not way of knowing which FW you got.
So I was supporting only the latest FW for that reason.

The MS tool don't let you choose a version so everyone is screwed but that's MS fault.

The last pre-BLE FW is still supported: 4.8.1923.0 (and probably a few after that) but if you don't have it already then you are screwed cause the MS tool will only update to latest BLE FW.

I only got one controller of each type, so I can only support latest since it's what I got access to.
I wish I had a controller with each FW version and try to figure a way to detect them etc, but it's not financially possible. (And it would be a pain to track them down too!)

**Reply by hipsofjw** [reply]
I don't know if this helps, but you can downgrade the firmware. I think the Xbox lets you do it through the normal controller menu, but on PC you do this:

"Open xbox accessories app then hit windows + R (run) and input this command:

xboxaccessories:\\firmwareupdate?legacyDowngrade=true

It should then give you option to revert."

Thanks for your detailed response! I get that this a bit of a niche issue, so I appreciate your effort.

**Reply by darthcloud** [reply]
ok I didn't know that, I'll take a look later.

**Reply by hipsofjw** [reply]
I just tried this to see if it works. It does, but with a big caveat. My controller was on 3.1.1221.0. I updated it to the latest firmware. Then I downgraded the firmware. Instead of letting me choose or take me back down to 4.8.1923.0,  it just reset it to 3.1.1221.0. So maybe this won't be THAT helpful.

**Reply by darthcloud** [reply]
Speculation, but I guess the controller got 2 FW slot, when you update it likely flash on the inactive slot, and then put that one active.

So the downgrade likely only switch the active slot to the previous one.

Would be interesting to dump the FW somehow and make an alternative application for flashing.

**Reply by hipsofjw** [reply]
Sadly, I don't think there's enough need to justify that. Instead, I'm just going to get one of those new 8bitdo USB sticks. They're pretty cheap and do BLE. I'll plug that into the Xbox Brook Adapter and hopefully that will solve my issue.

**Reply by hipsofjw** [reply]
> Sadly, I don't think there's enough need to justify that. Instead, I'm just going to get one of those new 8bitdo USB sticks. They're pretty cheap and do BLE. I'll plug that into the Xbox Brook Adapter and hopefully that will solve my issue.

Looks like this doesn't work. Not enough power through the USB to power both the Brooks and 8bitdo adapters at the same time. Even tried using a powered USB hub.
